### PR TITLE
Add inverse normal normalization (INT)

### LINKED ISSUE #155: Gaussianize feature values with inverse normalize transform (INT)
We used this transformation in a paper in preparation, it would be great to add this functionality to pycytominer to enable all future users access!

cc @jatinarora-upmc @shntnu

### PR DESCRIPTION
**Pull request by axiomcura**
# Description
This PR adds inverse normal normalization (INT) as a supported option in the `normalize` function. It also updates the CLI parameters so users can run INT directly in their analysis workflows.

Closes #155 

## What is the nature of your change?

- [ ] Bug fix (fixes an issue).
- [X] Enhancement (adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [X] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [X] New and existing unit tests pass locally with my changes.
- [X] I have added tests that prove my fix is effective or that my feature works.
- [X] I have deleted all non-relevant text in this pull request template.

## Summary by CodeRabbit

* **New Features**
  * Added an `inverse_normal` normalization option in the library and CLI.
  * Introduced `inverse_normal_n_quantiles` (default `1000`) to control quantile resolution for this method.
  * Exposed the corresponding inverse-normal transform through the public operations package.
* **Documentation**
  * Updated normalization documentation to describe `inverse_normal` and `inverse_normal_n_quantiles`.
* **Tests**
  * Added CLI and library tests verifying outputs match scikit-learn’s inverse-normal quantile mapping and that metadata columns remain unchanged.
  * Added unit tests for transformer correctness and `fit_transform` behavior.

### CODE REVIEW

**Comment by gwaybio** [comment]
@axiomcura - please let me know when you address the code rabbit comments and then I will review! Thanks!

**Comment by axiomcura** [comment]
I have attended Code rabbit's comments. Ready for your review @gwaybio

File: pycytominer/operations/transform.py [hunk 1]
```diff
@@ -338,3 +338,85 @@ def transform(self, X: pd.DataFrame, copy: Optional[bool] = None) -> pd.DataFram
...
+    -----
+    This transform is rank-based: values are first converted to quantile ranks,
+    then mapped to a normal distribution. The transformed values are therefore
+    normal scores, not the original raw measurements, and distances between raw
+    values are not preserved.
+    """
+
+    def __init__(
+        self,
+        n_quantiles=1000,
```
Comment (gwaybio) [inline pycytominer/operations/transform.py:379]: I'm not sure where, but can you add a docstring for `n_quantiles`? It should mirror, to the extent possible, it's behavior in `QuantileTransformer` https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.QuantileTransformer.html

I had a question, for example, about the appropriateness of this default value, and I found the answer in the sklearn docs

File: pycytominer/cli.py [hunk 2]
```diff
@@ -225,6 +226,7 @@ def normalize(
             spherize_center: Whether to center data before sphering.
             spherize_method: Spherize method to use.
             spherize_epsilon: Spherize epsilon parameter.
+            inverse_normal_n_quantiles: Number of quantiles for inverse normal normalization.
```
Comment (gwaybio) [inline pycytominer/cli.py]: consider elaborating on this, once you settle on language in the class above

File: pycytominer/normalize.py [hunk 3]
```diff
@@ -103,6 +104,9 @@ def normalize(
     spherize_epsilon : float, default 1e-6.
         The sphering (aka whitening) fudge factor parameter. The function only uses
         this variable if method = "spherize".
+    inverse_normal_n_quantiles : int, default 1000
```
Comment (gwaybio) [inline pycytominer/normalize.py]: same comment about bolstering docs and default behavior

File: tests/test_operations/test_transform.py [hunk 4]
```diff
@@ -98,3 +100,64 @@ def test_robust_mad():
     expected_result = data_df.shape[1]
 
     assert int(result) == expected_result
+
+
+def test_inverse_normal_transform_matches_quantile_transformer():
+    """Test that InverseNormalTransform wraps QuantileTransformer to produce normal quantile scores."""
+    scaler = InverseNormalTransform(n_quantiles=5)
```
Comment (gwaybio) [inline tests/test_operations/test_transform.py:107]: this is a good test, but I also wonder if you don't set n_quantiles (it stays at default) it results in the same allclose score.

From sklearn: 

n_quantiles int, default=1000 or n_samples

Number of quantiles to be computed. It corresponds to the number of landmarks used to discretize the cumulative distribution function. If n_quantiles is larger than the number of samples, n_quantiles is set to the number of samples as a larger number of quantiles does not give a better approximation of the cumulative distribution function estimator.

**Review by gwaybio — CHANGES_REQUESTED** [review]
this looks great Erik! One general comment that pops up in multiple places regarding bolstering docs. One other comments about testing default behavior

File: pycytominer/operations/transform.py (hunk 1 above)
Comment (axiomcura) [reply pycytominer/operations/transform.py:379]: Updated! the `n_quantiles` docstring to mirror `QuantileTransformer`, including its default value, CDF behavior, and automatic adjustment when it exceeds the number of samples.

File: pycytominer/cli.py (hunk 2 above)
Comment (axiomcura) [reply pycytominer/cli.py]: Done!

File: pycytominer/normalize.py (hunk 3 above)
Comment (axiomcura) [reply pycytominer/normalize.py]: I did some changes, let me know what you think!

File: tests/test_operations/test_transform.py (hunk 4 above)
Comment (axiomcura) [reply tests/test_operations/test_transform.py:107]: Ah.... Good point! I added a test using the default `n_quantiles` for both transformers, verifying that their outputs match and that `1000` is capped at the number of samples.

LMK what you think!

File: pycytominer/operations/transform.py [hunk 5]
```diff
@@ -338,3 +338,92 @@ def transform(self, X: pd.DataFrame, copy: Optional[bool] = None) -> pd.DataFram
...
+
+    Parameters
+    ----------
+    n_quantiles : int, default=1000
+        Number of quantiles to be computed. It corresponds to the number of landmarks
+        used to discretize the cumulative distribution function. If ``n_quantiles`` is
+        larger than the number of samples, it is set to the number of samples because
+        a larger number of quantiles does not improve the cumulative distribution
+        function estimate. The actual number used after fitting is available as
+        ``n_quantiles_``.
```
Comment (gwaybio) [inline pycytominer/operations/transform.py]:
```suggestion
        ``n_quantiles_``. See sklearn.preprocessing.QuantileTransformer for more details.
```

**Review by gwaybio — APPROVED** [review]
looks good! feel free to merge after addressing the final suggestion and all tests pass

**Comment by axiomcura** [comment]
Alrighty, thanks for the review @gwaybio. I'll merge after the tests are done !
